### PR TITLE
refactor(*): make "not found" error message user-friendly

### PR DESCRIPTION
**Pull request by selslack**
**What this PR does / why we need it**:
Improved error message for users experiencing error described in #1193.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -333,9 +333,15 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
...
 
 		originalInfo := original.Get(info)
+
+		// Resource exists in the current cluster state, but not in the current helm configuration
+		// See: https://github.com/helm/helm/issues/1193 for more info
 		if originalInfo == nil {
-			kind := info.Mapping.GroupVersionKind.Kind
-			return fmt.Errorf("no %s with the name %q found", kind, info.Name)
+			return fmt.Errorf(
+				"%s %q is not managed by Helm; delete the resource from the current cluster state to let Helm manage it",
```
Comment (bacongobbler) [inline pkg/kube/client.go]: Can we please change the error message to

> kind %s with the name %q already exists in the cluster. Please delete the resource before upgrading.

This error can also occur if another chart is managing the resource, so "is not managed by Helm" is not always true.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/kube/client.go [hunk 2]
```diff
@@ -333,9 +333,15 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 		}
 
 		originalInfo := original.Get(info)
+
+		// Resource exists in the current cluster state, but not in the current helm configuration
```
Comment (bacongobbler) [inline pkg/kube/client.go]: Please change this to

> The resource already exists in the cluster, but it wasn't defined in the previous release. In this case, we consider it to be a resource that is un-managed by Helm and error out, asking for the user to intervene.

Comment (selslack) [reply pkg/kube/client.go]: Done.

File: pkg/kube/client.go (hunk 1 above)
Comment (selslack) [reply pkg/kube/client.go]: Done.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**